app/controllers/application_controller.rb

```diff
@@ -107,7 +107,7 @@ def context
   end
 
   def current_sp
-    @current_sp ||= sp_from_sp_session || sp_from_request_id || sp_from_request_issuer_logout
```

Comment:
seems like we could move `sp_from_request_issuer_logout` to the SAML controller, since it depends on `saml_request`?

Reply:
I was trying to untangle a bit, and wasn't sure where to start safely, but that's a good idea. Made an attempt in https://github.com/18F/identity-idp/pull/6091/commits/e45a5c3d6fa0984e944a533521a7489acec0cfbf

Fwiw, this implementation stems from https://github.com/18F/identity-idp/pull/3460 and https://github.com/18F/identity-idp/pull/3461, so my understanding is the only thing depending on `sp_from_request_issuer_logout` should be the redirect uris in unauthenticated SAML logout contexts